Sources/Sentry/SentrySamplingProfiler.cpp

```diff
@@ -21,7 +21,7 @@ namespace profiling {
             mach_port_t port;
             clock_serv_t clock;
             mach_timespec_t delaySpec;
-            std::shared_ptr<ThreadMetadataCache> cache;
+            ThreadMetadataCache *cache;
```

Comment:
Shouldn't this be a `std::unique_ptr<ThreadMetadataCache>` @indragiek ?

Reply:
It can't be because the pointer is managed by `SentrySamplingProfiler`. We can't copy a `unique_ptr` from `SentrySamplingProfiler` into this struct, it can only be moved (which we don't want to do, since that makes the original `unique_ptr` invalid). So I'm falling back to a regular pointer here and the contract is that `SentrySamplingProfiler` will keep that valid for as long as this thread exists.

Reply:
Ah ok, that's what I was really curious about, why we couldn't move it. Good to know!